### PR TITLE
Add param to Add-AzVMDataDisk

### PR DESCRIPTION
**Pull request by grizzlytheodore**
## Description
 Added parameter `-SourceResourceId` to `Add-AzVMDataDisk` cmdlet.

resolves https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/1421

## Mandatory Checklist

- Please choose the target release of Azure PowerShell. (⚠️**Target release** is a different concept from **API readiness**. Please click below links for details.)
  - [x] [General release](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Public preview](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Private preview](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Engineering build](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] No need for a release

- [x] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For SDK-based development mode, update `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`.
        * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. 
    * For autorest-based development mode, include the changelog in the PR description.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** adjust version of module manually in pull request

### CODE REVIEW

**Comment by NoriZC** [comment]
@grizzlytheodore could you please add a test case for this update?

File: src/Compute/Compute/help/Add-AzVMDataDisk.md [hunk 1]
```diff
@@ -17,16 +17,17 @@ Adds a data disk to a virtual machine.
 ```
 Add-AzVMDataDisk [-VM] <PSVirtualMachine> [[-Name] <String>] [[-VhdUri] <String>] [[-Caching] <CachingTypes>]
  [[-DiskSizeInGB] <Int32>] [-Lun] <Int32> [-CreateOption] <String> [[-SourceImageUri] <String>]
- [-DiskEncryptionSetId <String>] [-DeleteOption <String>] [-DefaultProfile <IAzureContextContainer>]
- [<CommonParameters>]
+ [-DiskEncryptionSetId <String>] [-DeleteOption <String>] [-SourceResourceId <String>]
+ [-DefaultProfile <IAzureContextContainer>] [-ProgressAction <ActionPreference>] [<CommonParameters>]
```
Comment (Sandido) [inline src/Compute/Compute/help/Add-AzVMDataDisk.md:21]: ProgressAction is added unnecessarily again , this happens often these days.

File: src/Compute/Compute/VirtualMachine/Config/AddAzureVMDataDiskCommand.cs [hunk 2]
```diff
@@ -144,6 +145,13 @@ public class AddAzureVMDataDiskCommand : ComputeClientBaseCmdlet
         [PSArgumentCompleter("Detach", "Delete")]
         public string DeleteOption { get; set; }
 
+        [Parameter(
+            Mandatory = false,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "ARM ID of snapshot or disk restore point from which to create a disk.")]
+        [ValidateNotNullOrEmpty]
+        public string SourceResourceId { get; set; }
```
Comment (Sandido) [inline src/Compute/Compute/VirtualMachine/Config/AddAzureVMDataDiskCommand.cs:153]: Should this be a string or a ResourceId type? Or some other guid.

**Review by Sandido — CHANGES_REQUESTED** [review]

File: src/Compute/Compute/VirtualMachine/Config/AddAzureVMDataDiskCommand.cs (hunk 2 above)
Comment (Sandido) [reply src/Compute/Compute/VirtualMachine/Config/AddAzureVMDataDiskCommand.cs:153]: Should be a string, I was thinking of then using that resource id in a ResourceIdentifier object (like in Get-AzVmss) in ps to help parse it, but that's not needed here.

**Review by Sandido — CHANGES_REQUESTED** [review]

File: src/Compute/Compute/help/Add-AzVMDataDisk.md (hunk 1 above)
Comment (grizzlytheodore) [reply src/Compute/Compute/help/Add-AzVMDataDisk.md:21]: @NoriZC  generating md file adds "-ProgressAction". 
Should I be removing them?

**Comment by grizzlytheodore** [comment]
@NoriZC, I added a test. 
Also, could you answer my question about "ProgressAction" parameter?

File: src/Compute/Compute/help/Add-AzVMDataDisk.md (hunk 1 above)
Comment (NoriZC) [reply src/Compute/Compute/help/Add-AzVMDataDisk.md:21]: This is a known issue caused by PlatyPS in PowerShell 7.4. You can either remove it in pr or we will remove it in our release process.

**Comment by NoriZC** [comment]
LGTM, waiting approval from Compute side

**Comment by grizzlytheodore** [comment]
we don't need to wait from our team's approval. 
We looked at it collectively, but this is a very simple change/PR

File: src/Compute/Compute/Compute.format.ps1xml [hunk 3]
```diff
@@ -409,6 +409,10 @@
                 <Label>SourceImage</Label>
                 <PropertyName>SourceImage</PropertyName>
               </ListItem>
+              <ListItem>
+                <Label>SourceResource</Label>
```
[the remark this answers is not in the document]
Comment (grizzlytheodore) [reply src/Compute/Compute/Compute.format.ps1xml:413]: this adds "SourceResource" property column when looking at looking at DataDisk list

**Review by Sandido — APPROVED** [review]

**Pull request merged**